### PR TITLE
chore: add CODEOWNERS

### PR DESCRIPTION
**Pull request by nathanielc**
Automatically request review from appropriate teams based on code changed.

### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

**Review by wolffcm — APPROVED** [review]

**Comment by nathanielc** [comment]
There are two layers to CODEOWNERS. First if an owner is specified then the review is automatically requested. Second we can optionally set up branch protections to require approvals from owners. For now I have not enabled any requirements of approval. 

As the standard process is to get an approval from all reviewers I don't feel the need to enforce that explicitly yet.

Maybe once we have a few days with this enabled we can revisit.

**Pull request merged**
